File: Java/Assignment4_3.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name   : Assignment4_3.java
// Description : write a program which accept number form user and display its  non factors in decreasing order.
// Author      : Aniket Devaba Markad                        
// Date        : 6/12/2025 
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Number
{
    private int iNo;

    Number(int a)
    {
        this.iNo = a;

        if(iNo < 0)
    {
        iNo = -iNo;
    }
    
    }
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name : DisplayNonFactors
// Description   : This function is used to display the Non factors in decresing order. 
// Author        : Aniket Devaba Markad 
// Date          : 6/12/2025 
// 
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    public void DisplayNonFactors()
    {
        int iCnt = 0;

        System.out.println("Non Factors of "+ iNo + "  in decresing order is :");
        for(iCnt = iNo; iCnt >= 1;iCnt --)
        {
            if(iNo % iCnt != 0)
            {
                System.out.println(iCnt );
            }
        }
    
    }

}

public class Assignment4_3
{
    public static void main (String a[])
    {
        int iValue = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("enter the number : ");
        iValue = sobj.nextInt();

        Number nobj = new Number(iValue);

        nobj.DisplayNonFactors();

    }
}

File: CPP/Assignment4_3.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//  File name   : Assignment4_3.cpp
//  Description : Write a program which accept number from user and display non factors of that number.
//  Author      : Aniket Devaba Markad
//  Date        : 10/12/2025
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include <iostream>
using namespace std;

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//  Function Name : DisplayNonFactor
//  Description   : This function is used to display non factors of that number.
//  Input         : int
//  Output        : void
//  Author        : Aniket Devaba Markad
//  Date          : 10/12/2025
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

void DisplayNonFactor(int iNo)
{
    int iCnt = 0;

    if(iNo < 0)
    {
        iNo = -iNo;
    }

    for(iCnt = 1; iCnt < iNo; iCnt++)
    {
        if((iNo % iCnt) != 0)
        {
            cout << iCnt << " is non factor" << endl;
        }
    }
}

int main()
{
    int iValue = 0;

    cout << "Enter number: ";
    cin >> iValue;

    DisplayNonFactor(iValue);

    return 0;
}

// Time Complexity : O(N)
